### PR TITLE
Fix no issue alert visibility

### PR DESCRIPTION
**Pull request by Mottie**
## Summary

- _(Summarize the changes that have been made to the platform)_
  > The same alert was showing when no contestable issues are loaded vs an error accessing the contestable issues endpoint was encountered. This PR separates the alert error message (per original design) to show the appropriate alert. 
- _(If bug, how to reproduce)_
- _(What is the solution, why is this the solution)_
  > Implementation did not match design
- _(Which team do you work for, does your team own the maintenance of this component?)_
  > Benefits Decision Reviews
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)

[#73525](https://github.com/department-of-veterans-affairs/va.gov-team/issues/73525)

## Testing done

- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

|  |  |
| ------ | ----- |
| API error | api-error |
| No issues loaded | no-loaded |

## What areas of the site does it impact?

Decision Reviews

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/appeals/shared/components/ContestableIssues.jsx [hunk 1]
```diff
@@ -203,7 +206,8 @@ const ContestableIssues = props => {
   return (
     <>
       <div name="eligibleScrollElement" />
-      {showNoIssues && <NoIssuesLoadedAlert />}
+      {showApiFailure && <ApiFailureAlert />}
```
Comment (rocio-desantiago) [inline src/applications/appeals/shared/components/ContestableIssues.jsx:209]: Does this work like an if else statement where if the first condition isn't met, it checks the next boolean?

File: src/applications/appeals/shared/content/contestableIssues.jsx [hunk 2]
```diff
@@ -69,8 +69,16 @@ MaxSelectionsAlert.propTypes = {
...
+export const MessageAlert = ({
+  title,
+  message,
+  headerLevel = 3,
+  errorKey,
+  errorReason,
+  classes = 'vads-u-margin-bottom--2',
+}) => {
   const wrapAlert = useRef(null);
+  const Header = `h${headerLevel}`;
```
Comment (rocio-desantiago) [inline src/applications/appeals/shared/content/contestableIssues.jsx:81]: Nifty

File: src/applications/appeals/shared/content/contestableIssues.jsx [hunk 3]
```diff
@@ -84,74 +92,77 @@ export const NoIssuesLoadedAlert = () => {
...
-    event: 'visible-alert-box',
-    'alert-box-type': 'error',
-    'alert-box-heading': title,
-    'error-key': 'no_issues_selected',
-    'alert-box-full-width': false,
-    'alert-box-background-only': false,
-    'alert-box-closeable': false,
-    'reason-for-alert': 'Missing eligible issues',
-  });
+  const classes = `vads-u-margin-${inReviewMode ? 'y' : 'bottom'}--2`;
```
Comment (rocio-desantiago) [inline src/applications/appeals/shared/content/contestableIssues.jsx:155]: is the default for `vads-u-margin-y` a margin of 0?

**Review by rocio-desantiago — APPROVED** [review]
Just have some points of clarification

File: src/applications/appeals/shared/components/ContestableIssues.jsx (hunk 1 above)
Comment (Mottie) [reply src/applications/appeals/shared/components/ContestableIssues.jsx:209]: In this pattern, if the first value is truthy, the component after the `&&` is rendered; if the value is falsy, the line is ignored.

File: src/applications/appeals/shared/content/contestableIssues.jsx (hunk 2 above)
Comment (Mottie) [reply src/applications/appeals/shared/content/contestableIssues.jsx:81]: Thanks! :D

File: src/applications/appeals/shared/content/contestableIssues.jsx (hunk 3 above)
Comment (Mottie) [reply src/applications/appeals/shared/content/contestableIssues.jsx:155]: Default class is `vads-u-margin-bottom--2`. On the review & submit page, while in review mode (not edit mode), the class switches to `vads-u-margin-y--2`.

**Pull request merged**